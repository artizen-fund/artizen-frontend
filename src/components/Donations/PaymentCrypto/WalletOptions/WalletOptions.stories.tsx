import WalletOptions from './'

export default {
  title: 'donations/WalletOptions',
  component: WalletOptions,
  argTypes: {},
}

export const WalletOptionsComponent = (props: any) => <WalletOptions {...props} />
