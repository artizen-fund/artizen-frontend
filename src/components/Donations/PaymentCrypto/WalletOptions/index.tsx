import styled from 'styled-components'
import { rgba, assetPath } from '@lib'
import { palette, typography } from '@theme'

interface IWalletOptions {
  connectWallet: (s: string) => void
}

const WalletOptions = ({ connectWallet }: IWalletOptions) => {
  return (
    <Wrapper>
      <WalletOption onClick={() => connectWallet('metamask')}>
        <img src={assetPath('/assets/metamask.svg')} alt="connect with MetaMask" />
        <OptionTitle>MetaMask</OptionTitle>
        <OptionSubtitle>Connect via your browser</OptionSubtitle>
      </WalletOption>
      <WalletOption onClick={() => connectWallet('walletconnect')}>
        <img src={assetPath('/assets/walletConnect.svg')} alt="connect with WalletConnect" />
        <OptionTitle>WalletConnect</OptionTitle>
        <OptionSubtitle>Choose an alternative wallet</OptionSubtitle>
      </WalletOption>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  position: relative;
  grid-area: walletOptions;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  &:after {
    position: absolute;
    left: 50%;
    top: 0;
    width: 0.5px;
    height: 100%;
    background-color: ${rgba(palette.stone)};
    content: ' ';
  }
`

const WalletOption = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 10px;
  cursor: pointer;
  img {
    margin-bottom: 15px;
  }
`

const OptionTitle = styled.h3`
  ${typography.title.l4}
`
const OptionSubtitle = styled.h4`
  ${typography.label.l2}
`

export default WalletOptions
