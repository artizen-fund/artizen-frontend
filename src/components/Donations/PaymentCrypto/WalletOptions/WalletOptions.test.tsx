import renderer from 'react-test-renderer'
import WalletOptions from './'

describe('WalletOptions', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<WalletOptions />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
