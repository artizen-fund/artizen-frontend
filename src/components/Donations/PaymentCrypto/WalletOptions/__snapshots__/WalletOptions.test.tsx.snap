// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletOptions renders correctly 1`] = `
<div
  className="sc-268575ae-0 gjfjNd"
>
  <div
    className="sc-268575ae-1 jwnmTS"
    onClick={[Function]}
  >
    <img
      alt="connect with MetaMask"
      src="/assets/metamask.svg"
    />
    <h3
      className="sc-268575ae-2 leZytT"
    >
      MetaMask
    </h3>
    <h4
      className="sc-268575ae-3 bZcDbE"
    >
      Connect via your browser
    </h4>
  </div>
  <div
    className="sc-268575ae-1 jwnmTS"
    onClick={[Function]}
  >
    <img
      alt="connect with WalletConnect"
      src="/assets/walletConnect.svg"
    />
    <h3
      className="sc-268575ae-2 leZytT"
    >
      WalletConnect
    </h3>
    <h4
      className="sc-268575ae-3 bZcDbE"
    >
      Choose an alternative wallet
    </h4>
  </div>
</div>
`;
